api/src/Entity/MaterialList.php

```diff
@@ -42,6 +42,11 @@ class MaterialList extends BaseEntity implements BelongsToCampInterface, CopyFro
     /**
      * The items that are part of this list.
      */
+    #[Assert\Count(
+        exactly: 0,
+        exactMessage: 'It\'s not possible to delete a material list as long as it has items linked to it.',
```

Comment:
Do we already have a concept for api error messaging?

Reply:
We have #2365 which works towards displaying the validation messages from the backend.
We also have the more general epic issue #1277.